File: langgraph/result/qwen/7b-eq2js/21.js
function normalizeValues(numbers) {
    const minValue = Math.min(...numbers);
    const maxValue = Math.max(...numbers);
    const rangeValue = maxValue - minValue;

    const normalizedList = [];
    for (const number of numbers) {
        const normalizedNumber = (number - minValue) / rangeValue;
        normalizedList.push(normalizedNumber);
    }

    return normalizedList;
}
// method name: rescaleToUnit -> normalizeValues
// parameter name: numbers -> values
// variable name: minNumber -> minValue, maxNumber -> maxValue
// introduced rangeValue to store the difference between largest and smallest values
// replaced list comprehension with a for loop to append each normalized value

File: langgraph/data/semantic_equiv/21.py
from typing import List


def normalize_values(values: List[float]) -> List[float]:
    smallest = min(values)
    largest = max(values)
    range_value = largest - smallest
    normalized = [((value - smallest) / range_value) for value in values]
    return normalized

# method name: rescale_to_unit -> normalize_values
# parameter name: numbers -> values
# variable name: min_number -> smallest, max_number -> largest
# introduced new variable: range_value to store the difference between largest and smallest
# kept list comprehension for normalization but used new variable names